spark-commons/src/test/scala/za/co/absa/spark/commons/errorhandling/implementations/ErrorHandlingFilterRowsWithErrorsTest.scala

```diff
@@ -87,4 +88,18 @@ class ErrorHandlingFilterRowsWithErrorsTest extends AnyFunSuite with SparkTestBa
 
     assert(result == expected)
   }
+
+  test("errorColumnType should return a BooleanType") {
```

Comment:
These tests only kind of duplicate the implementation. I would try to add one errorcolumn's (`ErrorColumn.column`) column to an (empty) dataframe and check it's structure.
And in the other case apply the aggregation function to an the empty DataFrame and do the same.

Reply:
```scala
    val emptyDf = spark.emptyDataFrame
    val errorColumn: ErrorColumn = ErrorHandlingFilterRowsWithErrors.putErrorToColumn(???) // just add some error submission)
    val testDf = emptyDf.withColumn('error_column', errorColumn.column)
    // assert testDf schema to the ErrorHandlingFilterRowsWithErrors.errorColumnType
   // similarl with errorColumnAggregationType - well compare that the schema of the empty Df and testDf did not change after putErrorsWithGrouping
```